bundles/org.eclipse.passage.lic.jetty/src/org/eclipse/passage/lic/internal/jetty/JettyHandler.java

```diff
@@ -49,71 +47,32 @@ public JettyHandler(Function<NetRequest, NetResponse> handler) {
 
 	@Override
 	public boolean handle(Request request, Response response, Callback callback) throws Exception {
-		response.setStatus(200);
-		long contentLength = -1;
-		for (HttpField field : request.getHeaders()) {
-			HttpHeader header = field.getHeader();
-			if (header == null) {
-				continue;
-			}
-			switch (header) {
-			case CONTENT_LENGTH:
-				response.getHeaders().add(field);
-				contentLength = field.getLongValue();
-				break;
-			case CONTENT_TYPE:
-				response.getHeaders().add(field);
-				break;
-			case TRAILER:
-				response.setTrailersSupplier(HttpFields.build());
-				break;
-			case TRANSFER_ENCODING:
-				contentLength = Long.MAX_VALUE;
-				break;
-			}
-		}
-		if (contentLength > 0) {
-			//FIXME: AF: here we should somehow consider our "handler" function
-			Content.copy(request, response, Response.newTrailersChunkProcessor(response), callback);
-		} else {
-			callback.succeeded();
-		}
+		NetResponse netResponse = handler.apply(new JettyRequest(request));
+		write(netResponse, request, response, callback);
 		return true;
 
 	}
 
-//	@Override
-//	public void handle(String target, Request request, HttpServletRequest wrapper, HttpServletResponse envelope)
-//			throws IOException, ServletException {
-//		write(response(wrapper), envelope);
-//		request.setHandled(true);
-//	}
-
-	private NetResponse response(Request request) {
-		return handler.apply(new JettyRequest(request));
-	}
-
-	private void write(NetResponse response, HttpServletResponse envelope) throws IOException {
-		envelope.setContentType(response.contentType().contentType());
-		envelope.setCharacterEncoding("UTF-8"); //$NON-NLS-1$
-		if (response.failed()) {
-			envelope.sendError(response.error().code(), response.error().message());
-			return;
-		}
-		envelope.setStatus(HttpServletResponse.SC_OK);
-		if (response.carriesPayload()) {
-			byte[] payload;
-			try {
-				payload = response.payload();
-			} catch (Exception e) {
-				throw new IOException(e);
-			}
-			envelope.setContentLength(payload.length);
-			try (PrintWriter out = envelope.getWriter()) {
-				out.write(new String(payload, Charset.forName("UTF-8"))); //$NON-NLS-1$
-				out.flush();
+	private void write(NetResponse netResponse, Request request, Response response, Callback callback)
+			throws IOException {
+		if (netResponse.failed()) {
+			Error error = netResponse.error();
+			Response.writeError(request, response, callback, error.code(), error.message());
```

Comment:
spare unnecessary scope: if response is failed, just do what it needs to be done and then `return`, no need for `else` statement